pkg/mattermost/mattermost.go

```diff
@@ -0,0 +1,208 @@
+	webSocketClient, _ := model.NewWebSocketClient4(webSocketURL, client.AuthToken)
+	webSocketClient.Listen()
+	go func() {
+		for {
+			select {
+			case event := <-webSocketClient.EventChannel:
+				handleMessage(event, b.AllowKubectl, b.ClusterName, b.ChannelName)
+			}
+		}
+	}()
+	return
+}
+
+// Check if Mattermost server is reachable
+func checkServerConnection() error {
+	if _, resp := client.GetOldClientConfig(""); resp.Error != nil {
+		return resp.Error
+	}
+	return nil
+}
+
+// Check if team exists in Mattermost
+func getBotTeam(teamName string) (*model.Team, error) {
+	botTeam, resp := client.GetTeamByName(teamName, "")
+	if resp.Error != nil {
+		return botTeam, resp.Error
+	}
+	return botTeam, nil
+}
+
+// Check if botkube user exists in Mattermost
+func getBotUser(teamID string) (*model.User, error) {
+	users, resp := client.AutocompleteUsersInTeam(teamID, BotName, "")
+	if resp.Error != nil {
+		return nil, resp.Error
+	}
+	return users.Users[0], nil
+}
+
+// Create channel if not present and add botkube user in channel
+func getBotChannel(channelName, botTeamID, botUserID string) (*model.Channel, error) {
+	// Checking if channel exists
+	botChannel, resp := client.GetChannelByName(channelName, botTeamID, "")
+	if resp.Error != nil {
+		// Creating channel
+		if botChannel, resp = client.CreateChannel(mmChannel(channelName, botTeamID)); resp.Error != nil {
+			return botChannel, resp.Error
+		}
+	}
+	logging.Logger.Info("Channel created successfully in Mattermost")
+
+	// Adding Botkube user to channel
+	client.AddChannelMember(botChannel.Id, botUserID)
+	return botChannel, nil
+}
+
+// Check incomming message and take action
+func handleMessage(event *model.WebSocketEvent, allowkubectl bool, clusterName, channelName string) {
+	// Check incomming message event type
+	if event.Event != model.WEBSOCKET_EVENT_POSTED {
+		return
+	}
+	post := model.PostFromJson(strings.NewReader(event.Data["post"].(string)))
+
+	// Check if message posted by botkube and has @botkube prefix
+	if post.UserId == botUser.Id || !(strings.HasPrefix(post.Message, "@"+BotName+" ")) {
+		return
+	}
+	inMessage := strings.TrimPrefix(post.Message, "@"+BotName+" ")
+
+	// Check where the message is posted
+	isAuthChannel := false
+	if event.Broadcast.ChannelId == botChannel.Id {
+		isAuthChannel = true
+	}
+
+	e := execute.NewDefaultExecutor(inMessage, allowkubectl, clusterName, channelName, isAuthChannel)
+	outMessage := e.Execute()
+	if len(outMessage) == 0 {
+		logging.Logger.Info("Invalid request. Dumping the response")
+		return
+	}
+	sendMessage("`"+outMessage+"`", post.Id, event.Broadcast.ChannelId)
+}
+
+// Send messages to Mattermost
+func sendMessage(msg, postID, channelID string) {
+	post := &model.Post{}
+	post.ChannelId = channelID
+	post.RootId = postID
+	// Create file if message is too large
+	if len(msg) >= 3990 {
+		res, resp := client.UploadFileAsRequestBody([]byte(msg), channelID, Logs)
```

Comment:
@mugdha-adhav it is not necessary that only logs messages would have size greater than 3990. Some responses like, `get pods --all-namespaces` can also be too long. We should set title of the file same as the command we are executing (just like we do it for slack)
https://github.com/infracloudio/botkube/blob/master/pkg/slack/slack.go#L136